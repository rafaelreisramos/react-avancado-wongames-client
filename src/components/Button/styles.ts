import styled, { css, DefaultTheme } from 'styled-components'

import { ButtonProps } from '.'

type ContainerProps = Pick<ButtonProps, 'size'>

const containerModifiers = {
  small: (theme: DefaultTheme) => css`
    height: 3rem;
    font-size: ${theme.font.sizes.xsmall};
    padding: ${theme.spacings.xxsmall};
  `,

  medium: (theme: DefaultTheme) => css`
    height: 4rem;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.medium};
  `,

  large: (theme: DefaultTheme) => css`
    height: 5rem;
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xlarge};
  `
}

export const Container = styled.button<ContainerProps>`
  ${({ theme, size }) => css`
    background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
    color: ${theme.colors.white};
    border: none;
    border-radius: ${theme.border.radius};

    ${!!size && containerModifiers[size](theme)};
  `}
`
