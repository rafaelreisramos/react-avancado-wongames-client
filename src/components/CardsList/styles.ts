import styled, { css } from 'styled-components'

export const Container = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.white};
    padding: ${theme.spacings.xsmall};
  `}
`

export const Card = styled.li`
  ${({ theme }) => css`
    list-style: none;
    display: flex;
    align-items: center;
    background: ${theme.colors.lightGray};
    padding: ${theme.spacings.xxsmall};
    color: ${theme.colors.black};
    height: 5rem;
    border-radius: calc(${theme.border.radius} / 2);

    &:not(:last-child) {
      margin-bottom: ${theme.spacings.xsmall};
    }

    > span {
      margin-left: ${theme.spacings.xxsmall};
    }
  `}
`
