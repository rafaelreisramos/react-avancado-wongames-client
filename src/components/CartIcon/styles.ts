import styled, { css } from 'styled-components'

export const Container = styled.div`
  ${({ theme }) => css`
    height: 2.4rem;
    width: 2.4rem;
    color: ${theme.colors.white};
    position: relative;
  `}
`

export const Badge = styled.span`
  ${({ theme }) => css`
    position: absolute;
    top: -0.4rem;
    right: -0.4rem;
    width: 1.6rem;
    height: 1.6rem;
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${theme.colors.secondary};
    color: ${theme.colors.white};
    border-radius: 50%;
    font-size: 1rem;
  `}
`
