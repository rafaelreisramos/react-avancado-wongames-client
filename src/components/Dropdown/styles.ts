import styled, { css } from 'styled-components'

export const Title = styled.div`
  ${({ theme }) => css`
    position: relative;
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    padding-right: ${theme.spacings.small};
    cursor: pointer;
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    position: absolute;
    right: 0;
    margin-top: ${theme.spacings.small};
    display: flex;
    flex-direction: column;
    background: ${theme.colors.white};
    color: ${theme.colors.black};

    &::before {
      content: '';
      position: absolute;
      border-right: 1.2rem solid transparent;
      border-left: 1.2rem solid transparent;
      border-bottom: 1.2rem solid ${theme.colors.white};
      top: -1.2rem;
      right: 2.4rem;
    }
  `}
`

type ContainerProps = {
  isOpen?: boolean
}

const containerModifiers = {
  open: () => css`
    opacity: 1;
    pointer-events: auto;
    transform: translateY(0);
  `,

  close: () => css`
    opacity: 0;
    pointer-events: none;
    transform: translateY(-2rem);
  `
}

export const Container = styled.div<ContainerProps>`
  ${({ theme, isOpen }) => css`
    position: relative;
    width: max-content;

    ${Content} {
      transition: transform 0.2s ease-in, opacity ${theme.transition.default};

      ${isOpen && containerModifiers.open()}
      ${!isOpen && containerModifiers.close()}
    }
  `}
`
