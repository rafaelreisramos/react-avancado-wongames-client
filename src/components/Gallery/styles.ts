import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Container = styled.div`
  ${({ theme }) => css`
    .slick-prev,
    .slick-next {
      display: block;
      width: 2.5rem;
      height: 2.5rem;
      padding: 0;
      color: ${theme.colors.white};
      cursor: pointer;
      position: absolute;
      top: 50%;
      transform: translate(0, -50%);

      &.slick-disabled {
        visibility: hidden;
      }
    }

    .slick-prev {
      left: -${theme.spacings.xxlarge};
    }

    .slick-next {
      right: -${theme.spacings.xxlarge};
    }

    .slick-slide > div {
      margin: 0 ${theme.spacings.xsmall};
      cursor: pointer;
    }

    .slick-list {
      margin: 0 -${theme.spacings.xsmall};
    }

    ${media.lessThan('huge')`
      overflow-x: hidden;
    `}
  `}
`

type ModalProps = {
  isOpen: boolean
}

const modalModifiers = {
  open: () => css`
    opacity: 1;
  `,

  close: () => css`
    opacity: 0;
    pointer-events: none;
  `
}
export const Modal = styled.div<ModalProps>`
  ${({ theme, isOpen }) => css`
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    background: rgba(0, 0, 0, 0.7);
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: ${theme.layers.modal};
    transition: opacity ${theme.transition.default};

    ${isOpen && modalModifiers.open()}
    ${!isOpen && modalModifiers.close()}
  `}
`

export const Close = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    position: absolute;
    left: 0;
    top: 0;
    cursor: pointer;
    width: 100%;
    height: 100%;
    text-align: right;
  `}
`

export const Content = styled.div`
  max-width: min(120rem, 100%);
  max-height: 80rem;
`
