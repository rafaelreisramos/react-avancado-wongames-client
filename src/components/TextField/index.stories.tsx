import { Story, Meta } from '@storybook/react/types-6-0'
import { Email } from '@styled-icons/material-outlined'

import TextField, { TextFieldProps } from '.'

export default {
  title: 'Form/TextField',
  component: TextField,
  args: {
    id: 'email',
    label: 'E-mail',
    labelFor: 'email',
    icon: <Email />,
    initialValue: '',
    placeholder: 'john.doe@gmail.com'
  },
  argTypes: {
    onInput: { action: 'typed text' },
    icon: {
      type: ''
    }
  }
} as Meta

export const Default: Story<TextFieldProps> = (args) => (
  <div style={{ maxWidth: 300, padding: 15 }}>
    <TextField {...args} />
  </div>
)

export const withError: Story<TextFieldProps> = (args) => (
  <div style={{ maxWidth: 300, padding: 15 }}>
    <TextField {...args} />
  </div>
)

withError.args = {
  error: 'Ops... something is wrong'
}
