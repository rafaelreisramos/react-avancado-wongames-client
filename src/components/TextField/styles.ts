import styled, { css, DefaultTheme } from 'styled-components'

import { TextFieldProps } from '.'

type IconPositionProps = Pick<TextFieldProps, 'iconPosition'>
type ContainerProps = Pick<TextFieldProps, 'disabled'> & { error?: boolean }

const containerModifiers = {
  disabled: (theme: DefaultTheme) => css`
    ${Label}, ${Input}, ${Icon} {
      cursor: not-allowed;
      color: ${theme.colors.gray};

      &::placeholder {
        color: currentColor;
      }
    }
  `,

  error: (theme: DefaultTheme) => css`
    ${InputBox} {
      border-color: ${theme.colors.red};
    }

    ${Label},
    ${Icon} {
      color: ${theme.colors.red};
    }
  `
}

export const Container = styled.div<ContainerProps>`
  ${({ theme, error, disabled }) => css`
    ${error && containerModifiers.error(theme)}
    ${disabled && containerModifiers.disabled(theme)}
  `}
`

export const Label = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.black};
    cursor: pointer;
  `}
`

export const InputBox = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    background: ${theme.colors.lightGray};
    padding: 0 ${theme.spacings.xsmall};
    border-radius: calc(${theme.border.radius} / 2);
    border: 0.2rem solid ${theme.colors.lightGray};

    &:focus-within {
      box-shadow: 0 0 0.5rem ${theme.colors.primary};
    }
  `}
`

export const Icon = styled.div<IconPositionProps>`
  ${({ theme, iconPosition }) => css`
    display: flex;
    width: 2.2rem;
    color: ${theme.colors.gray};
    order: ${iconPosition === 'right' ? 1 : 0};

    & > svg {
      width: 100%;
    }
  `}
`

export const Input = styled.input<IconPositionProps>`
  ${({ theme, iconPosition }) => css`
    border: 0;
    outline: none;
    background: transparent;
    width: 100%;
    padding: ${theme.spacings.xxsmall} 0;
    padding-${iconPosition}: ${theme.spacings.xsmall};
    color: ${theme.colors.black};
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.medium};
  `}
`

export const ErrorMessage = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.red};
    font-size: ${theme.font.sizes.xsmall};
  `}
`
