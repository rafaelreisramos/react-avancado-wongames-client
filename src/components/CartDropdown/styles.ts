import styled from 'styled-components'

import * as CartListStyles from 'components/CartList/styles'

export const Container = styled.div`
  ${CartListStyles.Container} {
    width: 56rem;
  }
`
