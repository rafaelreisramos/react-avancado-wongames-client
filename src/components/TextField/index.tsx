import React, { InputHTMLAttributes, useState } from 'react'
import * as S from './styles'

export type TextFieldProps = {
  label?: string
  labelFor?: string
  initialValue?: string
  onInput?: (value: string) => void
  icon?: React.ReactNode
  iconPosition?: 'left' | 'right'
  disabled?: boolean
  error?: string
} & InputHTMLAttributes<HTMLInputElement>

const TextField = ({
  label,
  labelFor = '',
  initialValue = '',
  onInput,
  icon,
  iconPosition = 'left',
  disabled = false,
  error,
  ...props
}: TextFieldProps) => {
  const [value, setValue] = useState(initialValue)

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = e.currentTarget.value
    setValue(newValue)

    !!onInput && onInput(newValue)
  }

  return (
    <S.Container disabled={disabled} error={!!error}>
      {!!label && <S.Label htmlFor={labelFor}>{label}</S.Label>}
      <S.InputBox>
        {!!icon && <S.Icon iconPosition={iconPosition}>{icon}</S.Icon>}
        <S.Input
          type="text"
          iconPosition={iconPosition}
          onChange={onChange}
          value={value}
          disabled={disabled}
          {...props}
        />
      </S.InputBox>
      {!!error && <S.ErrorMessage>{error}</S.ErrorMessage>}
    </S.Container>
  )
}

export default TextField
