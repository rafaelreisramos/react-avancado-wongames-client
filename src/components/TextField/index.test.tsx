import { render, screen, waitFor } from 'utils/test-utils'
import userEvent from '@testing-library/user-event'
import { Email } from '@styled-icons/material-outlined'

import TextField from '.'

describe('<TextField />', () => {
  it('should render with label', () => {
    render(<TextField label="label" name="field" />)

    expect(screen.getByLabelText('label')).toBeInTheDocument()
  })

  it('should render without label', () => {
    render(<TextField />)

    expect(screen.queryByLabelText(/label/i)).not.toBeInTheDocument()
  })

  it('should render with a placeholder', () => {
    render(<TextField placeholder="placeholder" />)

    expect(screen.getByPlaceholderText(/placeholder/i)).toBeInTheDocument()
  })

  it('should render with an icon', () => {
    render(<TextField icon={<Email data-testid="icon" />} />)

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render with an icon at right', () => {
    render(
      <TextField iconPosition="right" icon={<Email data-testid="icon" />} />
    )

    expect(screen.getByTestId('icon').parentElement).toHaveStyle({ order: 1 })
  })

  it(`should change it's value when typing`, async () => {
    const onInput = jest.fn()

    render(<TextField onInput={onInput} label="label" name="field" />)

    const input = screen.getByRole('textbox')
    const text = 'Input text'

    userEvent.type(input, text)
    await waitFor(() => {
      expect(input).toHaveValue(text)
      expect(onInput).toHaveBeenCalledTimes(text.length)
    })

    expect(onInput).toHaveBeenCalledWith(text)
  })

  it(`should not change it's value when disabled`, async () => {
    const onInput = jest.fn()

    render(<TextField onInput={onInput} label="label" name="field" disabled />)

    const input = screen.getByRole('textbox')
    expect(input).toBeDisabled()

    const text = 'Input text'

    userEvent.type(input, text)
    await waitFor(() => {
      expect(input).not.toHaveValue(text)
    })

    expect(onInput).not.toHaveBeenCalledWith()
  })

  it('should not be accessible with tab when disabled', () => {
    render(<TextField label="label" name="field" disabled />)

    expect(document.body).toHaveFocus()

    userEvent.tab()
    expect(screen.getByRole('textbox')).not.toHaveFocus()
  })

  it('should be accessible with tab', () => {
    render(<TextField label="label" name="field" />)

    expect(document.body).toHaveFocus()

    userEvent.tab()
    expect(screen.getByRole('textbox')).toHaveFocus()
  })

  it('should render with error', () => {
    const { container } = render(
      <TextField
        icon={<Email data-testid="icon" />}
        label="label"
        name="field"
        error="Error message"
      />
    )

    expect(screen.getByText('Error message')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
