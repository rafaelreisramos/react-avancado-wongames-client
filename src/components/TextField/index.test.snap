// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render with error 1`] = `
.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 .c3 {
  border-color: #ff6347;
}

.c0 .c1,
.c0 .c5 {
  color: #ff6347;
}

.c2 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #eaeaea;
  padding: 0 1.6rem;
  border-radius: calc(0.4rem / 2);
  border: 0.2rem solid #eaeaea;
}

.c4:focus-within {
  box-shadow: 0 0 0.5rem #f231a5;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2.2rem;
  color: #8f8f8f;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c6 > svg {
  width: 100%;
}

.c8 {
  border: 0;
  outline: none;
  background: transparent;
  width: 100%;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c9 {
  color: #ff6347;
  font-size: 1.2rem;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
    for="field"
  >
    label
  </label>
  <div
    class="c3 c4"
  >
    <div
      class="c5 c6"
    >
      <svg
        aria-hidden="true"
        class="c7"
        data-testid="icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
        />
      </svg>
    </div>
    <input
      class="c8"
      type="text"
      value=""
    />
  </div>
  <span
    class="c9"
  >
    Error message
  </span>
</div>
`
